pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java

```diff
@@ -0,0 +1,206 @@
+  @ApiOperation(value = "Gets leaders for all tables", notes = "Gets leaders for all tables")
+  public LeadControllerResponse getLeadersForAllTables() {
+    Map<String, LeadControllerEntry> leadControllerEntryMap = new LinkedHashMap<>();
+    HelixManager helixManager = _pinotHelixResourceManager.getHelixZkManager();
+    boolean isLeadControllerResourceEnabled;
+    try {
+      isLeadControllerResourceEnabled = LeadControllerUtils.isLeadControllerResourceEnabled(helixManager);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER,
+          "Exception when checking whether lead controller resource is enabled or not.",
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    // Returns empty map if lead controller resource is disabled.
+    if (!isLeadControllerResourceEnabled) {
+      return new LeadControllerResponse(false, leadControllerEntryMap);
+    }
+
+    ExternalView leadControllerResourceExternalView = getLeadControllerResourceExternalView(helixManager);
+    for (int partitionId = 0; partitionId < Helix.NUMBER_OF_PARTITIONS_IN_LEAD_CONTROLLER_RESOURCE; partitionId++) {
+      String partitionName = LeadControllerUtils.generatePartitionName(partitionId);
+      String participantInstanceId =
+          getParticipantInstanceIdFromExternalView(leadControllerResourceExternalView, partitionName);
+      if (participantInstanceId == null) {
+        continue;
+      }
+      leadControllerEntryMap
+          .putIfAbsent(partitionName, new LeadControllerEntry(participantInstanceId, new ArrayList<>()));
+    }
+
+    // Assigns all the tables to the relevant partitions.
+    List<String> tableNames = _pinotHelixResourceManager.getAllTables();
+    for (String tableName : tableNames) {
+      String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+      int partitionId = LeadControllerUtils.getPartitionIdForTable(rawTableName);
+      String partitionName = LeadControllerUtils.generatePartitionName(partitionId);
+      LeadControllerEntry leadControllerEntry = leadControllerEntryMap.get(partitionName);
+      leadControllerEntry.getTableNames().add(tableName);
+    }
+    return new LeadControllerResponse(true, leadControllerEntryMap);
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}")
+  @ApiOperation(value = "Gets leader for a given table", notes = "Gets leader for a given table")
+  public LeadControllerResponse getLeaderForTable(
+      @ApiParam(value = "Table name", required = true) @PathParam("tableName") String tableName) {
+    Map<String, LeadControllerEntry> leadControllerEntryMap = new HashMap<>();
+    HelixManager helixManager = _pinotHelixResourceManager.getHelixZkManager();
+    boolean isLeadControllerResourceEnabled;
+    try {
+      isLeadControllerResourceEnabled = LeadControllerUtils.isLeadControllerResourceEnabled(helixManager);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER,
```

Comment:
include `e` in ControllerApplicationException

Reply:
Done.